thestr4ng3r/retdec-r2plugin
# Emit RAnnotatedCode to cutter

**Detailed description**


In this PR, RAnnotatedCode is emitted from the decompileAt() function called by Cutter. When a function is not found at an offset, the following message is shown. I decided to have this message seeing the corresponding message given by the R2Ghidra plugin.
![Screenshot from 2020-06-02 12-04-00](https://user-images.githubusercontent.com/18501167/83487726-2b4bef80-a4c9-11ea-94a4-5cd1a3b1024f.png)
![Screenshot from 2020-06-02 11-00-47](https://user-images.githubusercontent.com/18501167/83487730-2d15b300-a4c9-11ea-81e1-e8f6bf4550e3.png)


**Test plan**


1. Get PR #[2227](https://github.com/radareorg/cutter/pull/2227) and the latest radare2 version before compiling this PR.
2. Open the decompiler widget and make sure that the correct decompiled code is being shown while you select offsets from functions and other widgets.
3. Make sure that the decompiler widget is seeking the correct address while clicking on functions(and other offsets) in the decompiler widget.
4. Make sure that invalid offsets are not causing Segmentation Fault.
5. Check the code and make sure that I have not missed any exceptional cases that will cause errors.

**Notes**
Merge all these three PRs together to ensure smooth functioning of all decompilers.
Refactored decompiler widget and R2Dec plugin: [Cutter #2227 ](https://github.com/radareorg/cutter/pull/2227)
Refactored R2Ghidra plugin that emits RAnnotatedCode: [R2Ghidra-dec #112](https://github.com/radareorg/r2ghidra-dec/pull/112)
Refactored retdec plugin that emits RAnnotatedCode:[ Florian's fork #1](https://github.com/thestr4ng3r/retdec-r2plugin/pull/1), after this PR is verified, I will send it to retdec's repo.
